Update isend/irecv warning messages for nccl

H-Huang opened the pull request:
Summary: nccl backend does not support `tag` as mentioned in https://github.com/pytorch/pytorch/issues/94819. Adding a note in the documentation for it.

Example:

<img width="888" alt="image" src="https://user-images.githubusercontent.com/14858254/220464900-094c8063-797a-4bdc-8e25-657f17593fe9.png">


Differential Revision: D43475756

awgu commented on torch/distributed/distributed_c10d.py:
```diff
@@ -1227,6 +1227,9 @@ def isend(tensor: torch.Tensor, dst: int, group: Optional[ProcessGroup] = None,
         behavior.
 
+    .. warning::
+        ``tag`` and tensor ordering is not supported with the NCCL backend.
```
Should "is" be "are"? 🤔

H-Huang replied:
updated to remove "tensor ordering" part

H-Huang committed:
Update isend/irecv warning messages for nccl (#95236)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/95236

nccl backend does not support `tag` as mentioned in https://github.com/pytorch/pytorch/issues/94819. Adding a note in the documentation for it.

Test Plan: sandcastle

Reviewed By: rohan-varma, awgu

Differential Revision: D43475756

fbshipit-source-id: 76106dc76170b087c72f180375f7b24e487d7a9e
